tests/semantic_analyser.cpp

```diff
@@ -2702,6 +2703,16 @@ TEST(semantic_analyser, string_size)
   ASSERT_EQ(var_assign->var->type.GetField(0).type.GetSize(), 6UL);
 }
 
+TEST(semantic_analyser, call_nsecs)
+{
+  test("BEGIN { $ns = nsecs(); }", 0);
+  test("BEGIN { $ns = nsecs(boot); }", 0);
+  // The CI kernel doesn't support tai related helper function
+  // test("BEGIN { $ns = nsecs(tai); }", 0);
```

Comment:
It should be possible to add a conditional test for this:
```cpp
TEST(semantic_analyser, call_nsecs_tai)
{
  BPFfeature feature;
  if (!feature.has_helper_ktime_get_tai_ns())
    GTEST_SKIP();

  test("BEGIN { $ns = nsecs(tai); }", 0);
}
```

Reply:
It'd be better to use `MockBPFfeature` here - the unit tests shouldn't depend on the environment.

@chengshuyi you'll need to add your new feature to MockBPFfeature, like I am doing here: https://github.com/iovisor/bpftrace/pull/2645/files#diff-aa3b9f9492e765d109d2fd769d283c0fd6b0a76ffcd11cd02054041ae0598948R120

The semantic analyser tests will run with this feature enabled once you do this.

(Independently, I'm refactoring this mocking so that we won't need to remember to update this separate class every time a test requires a new feature)

Reply:
Update to patch: [add semantic analyser test for nsecs()](https://github.com/iovisor/bpftrace/pull/2619/commits/f899a1523da41757255ccb03945016f66db7bcc5). Thanks.